octopusdeploy_framework/resource_tag.go

```diff
@@ -110,17 +110,18 @@ func (t *tagTypeResource) Update(ctx context.Context, req resource.UpdateRequest
 	tagSetID := data.TagSetId.ValueString()
 	tagSetSpaceID := data.TagSetSpaceId.ValueString()
 
-	tflog.Info(ctx, fmt.Sprintf("updating tag (%s)", data.ID))
+	tflog.Info(ctx, fmt.Sprintf("updating tag (%s)", state.ID))
 
 	// if the tag is reassigned to another tag set
 	if !data.TagSetId.Equal(state.TagSetId) {
 		sourceTagSetID, destinationTagSetID := state.TagSetId.ValueString(), data.TagSetId.ValueString()
-		sourceTagSetSpaceID, destinationTagSetSpaceID := state.TagSetSpaceId.ValueString(), data.TagSetSpaceId.ValueString()
+		targetSpaceId := util.Ternary(data.TagSetSpaceId.ValueString() == "", data.TagSetSpaceId, state.TagSetSpaceId)
```

Comment:
Use the configured spaceId otherwise fall back to the data (only set if the tag spaceId is set)